### PR TITLE
Personal view real data

### PR DESCRIPTION
**Pull request by jKostet**
moi

- [x] omien puheluiden määrä
- [x] omien puheluiden keston keskiarvo
- [x] omien jälkikirjausten keston keskiarvo
- [x] korjaa ^ laskeminen backend
- [x] jonon pituus
- [x] oma status
- [x] oma väri
- [x] muiden status
- [x] kirjoita RSpec-testejä
- [x] kirjoita Protractor-testejä

### CODE REVIEW

File: app/controllers/contacts_controller.rb [hunk 1]
```diff
@@ -6,8 +6,8 @@ def init
     time = Time.zone.now
     @start_time = time.beginning_of_day
     @end_time = time.end_of_day
-    @team_name = 'Helpdesk'
-    @contacts_service = ContactsService.new(@team_name, @start_time, @end_time)
+    @team = Team.find_by(name: 'Helpdesk')
```
Comment (siiptuo) [inline app/controllers/contacts_controller.rb]: Small thing but `find_by_name` looks prettier.

File: app/assets/javascripts/screen/screenService.js [hunk 2]
```diff
@@ -13,6 +13,20 @@ angular.module('ocWebGui.screen.service', ['ngResource', 'ocWebGui.filterpanel']
             })
             .map(function (agent) {
               agent.created_at = new Date(agent.created_at);
+
+              switch (agent.status) {
```
Comment (siiptuo) [inline app/assets/javascripts/screen/screenService.js]: Duplicated logic in `ScreenController`.

File: app/controllers/personal_status_controller.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# API for OC Agent status data
+class PersonalStatusController < ApplicationController
+  def index
+    user = User.find_by(id: session[:user_id])
```
Comment (siiptuo) [inline app/controllers/personal_status_controller.rb]: `current_user` from `ApplicationController` can be used here.

File: app/controllers/personal_status_controller.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    agent_id = user.agent_id
+    if agent_id.nil?
+      render json: {}, status: 404
+      return
+    end
+    agent = Agent.find_by(id: agent_id)
```
Comment (siiptuo) [inline app/controllers/personal_status_controller.rb]: Doesn't `user.agent` do the same thing?

File: app/controllers/personal_status_controller.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    if agent.nil?
+      render json: {}, status: 403
+      return
+    end
+
+    contacts_service = ContactsService.new(Agent.find_by(id: agent_id), Time.zone.now.beginning_of_day, Time.zone.now.end_of_day)
```
Comment (siiptuo) [inline app/controllers/personal_status_controller.rb]: We have agent in a variable already.

File: app/assets/javascripts/personal/personalController.js [hunk 6]
```diff
@@ -8,6 +8,41 @@ angular.module('ocWebGui.personal', ['ui.router'])
...
+      });
+
+      Queue.query(function (queue) {
+        vm.queue = queue;
+      });
+
+      Personal.get(function(data) {
+        vm.myCalls_count = data.answered_calls;
+        vm.myCalls_avg = data.average_call_duration;
+        vm.myAftercalls_avg = data.average_after_call_duration;
```
Comment (siiptuo) [inline app/assets/javascripts/personal/personalController.js]: `data` could simple be passed to template as it is.

File: app/assets/javascripts/personal/personalController.js [hunk 7]
```diff
@@ -8,6 +8,41 @@ angular.module('ocWebGui.personal', ['ui.router'])
...
   })
-  .controller('PersonalController', function () {
+  .controller('PersonalController', function (Agents, Queue, User, Personal, $interval, $scope) {
     var vm = this;
+
+    function fetchData() {
+      Agents.query(function (agents) {
+        vm.agents = agents;
+
+        var myAgent = agents.filter(function (agent) {
```
Comment (siiptuo) [inline app/assets/javascripts/personal/personalController.js]: `Array.find` could be used instead.

**Comment by siiptuo** [comment]
Only thing consistent in our CSS is `kebab-case`...

File: app/assets/javascripts/personal/_personal.html [hunk 8]
```diff
@@ -1,41 +1,31 @@
...
-                <li>
-                    <span class="color-square agent-status-color-green"></span>
-                    <span>Risto C</span>
-                </li>
-                <li>
-                    <span class="color-square agent-status-color-yellow"></span>
-                    <span>Pekka D</span>
+                <li ng-repeat="agent in personal.agents track by agent.id">
+                    <span class="color-square agent-status-color-{{agent.color}}"></span>
+                    <span class="agentName">{{ agent.first_name }}</span>
```
Comment (siiptuo) [inline app/assets/javascripts/personal/_personal.html]: Only first name displayed unlike in status view. Is a problem?

**Comment by siiptuo** [comment]
All agents are listed including the logged user him or herself.

File: app/assets/javascripts/personal/_personal.html (hunk 8 above)
Comment (banjohirvi) [inline app/assets/javascripts/personal/_personal.html]: Yep, will fix

**Pull request merged**